src/util/system.cpp

```diff
@@ -1038,6 +1042,39 @@ void runCommand(const std::string& strCommand)
 }
 #endif
 
+#ifdef HAVE_BOOST_PROCESS
+UniValue runCommandParseJSON(const std::string& str_command, const std::string& str_std_in)
+{
+    namespace bp = boost::process;
+
+    UniValue result_json;
+    bp::opstream stdin_stream;
+    bp::ipstream stdout_stream;
+
+    if (str_command.empty()) return UniValue::VNULL;
+
+    bp::child c(
+        str_command,
+        bp::std_out > stdout_stream,
+        bp::std_in < stdin_stream
+    );
+    if (!str_std_in.empty()) {
+        stdin_stream << str_std_in << std::endl;
+        stdin_stream.pipe().close();
```

Comment:
Valgrind warns here: `==22385== Warning: invalid file descriptor -1 in syscall close()` during the test with `std::in`.